lib/core/src/display/shape/text/text_field.rs

```diff
@@ -172,9 +159,22 @@ shared! { TextField
                 let cursor_with_line = cursor_ids.iter().map(|cursor_id| (*cursor_id,insertion));
                 self.edit_per_cursor(cursor_with_line);
             };
-            self.assignment_update(fonts).update_after_text_edit();
-            self.rendered.update_glyphs(&mut self.content,fonts);
-            self.rendered.update_cursor_sprites(&self.cursors, &mut self.content.full_info(fonts));
+            self.assignment_update().update_after_text_edit();
+            self.rendered.update_glyphs(&mut self.content);
+            self.rendered.update_cursor_sprites(&self.cursors, &mut self.content);
+        }
+
+        /// Do delete operation on text.
+        ///
+        /// For cursors with selection it will just remove the selected text. For the rest, it will
+        /// remove all content covered by `step`.
+        pub fn do_delete_operation(&mut self, step:Step) {
+            let content           = &mut self.content;
+            let selecting         = true;
+            let mut navigation    = CursorNavigation {content,selecting};
+            let without_selection = |c:&Cursor| !c.has_selection();
+            self.cursors.navigate_cursors(&mut navigation,step,without_selection);
+            self.edit("");
```

Comment:
After seeing this line I started to believe that the name `edit` is misleading.  `Edit` should take a lambda or an action. What does it mean to pass empty strign to edit function?